matrix.yaml

```diff
@@ -28,15 +28,11 @@ exclude:
   - LINUX_VER: "centos7"
     IMAGE_REPO: "citestwheel"
 
-  # exclude citestwheel and ci-wheel for cuda versions other than 11.8.0 and 12.0.1
+  # exclude citestwheel and ci-wheel for cuda versions other than 11.8.0, 12.0.1, and 12.2.2
   - CUDA_VER: "11.4.3"
     IMAGE_REPO: "citestwheel"
   - CUDA_VER: "11.4.3"
     IMAGE_REPO: "ci-wheel"
-  - CUDA_VER: "12.1.1"
-    IMAGE_REPO: "citestwheel"
-  - CUDA_VER: "12.1.1"
-    IMAGE_REPO: "ci-wheel"
```

Comment:
I think these exclusions should be updated to include `12.2.2`.

@vyasr can comment, but I think for wheels we want to build/test on the earliest CUDA images.

Reply:
I chose not to add these exclusions based on this statement in https://github.com/rapidsai/build-planning/issues/7:

> *Modify the [wheel shared-workflows](https://github.com/rapidsai/shared-workflows/tree/branch-24.02/.github/workflows) to use the new images in building wheels. These jobs [should be set up to continue on error](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idcontinue-on-error)*

Won't `ci-wheel` and `citestwheel` images with CUDA 12.2 be necessary to do that?

Reply:
I'll let Vyas comment. He knows the wheel stuff better than me.

At the very least, the quote you mentioned applies to the conda images. Not sure about the wheels.

Reply:
> I'll let Vyas comment.

hmmm ok, yeah I'll wait for that.

I guess those exclusions were also covering the fact that there aren't Ubuntu 18.04 `nvidia/cuda` images CUDA 12.1 ([DockerHub search](https://hub.docker.com/r/nvidia/cuda/tags?page=1&name=12.1.1-devel-ubuntu)) or 12.2 ([DockerHub search](https://hub.docker.com/r/nvidia/cuda/tags?page=1&name=12.2.2-devel-ubuntu)) on DockerHub, but there are for 12.0 ([DockerHub search](https://hub.docker.com/r/nvidia/cuda/tags?page=1&name=12.0.1-devel-ubuntu)).

https://github.com/rapidsai/ci-imgs/actions/runs/7277784010?pr=101

<img width="1176" alt="image" src="https://github.com/rapidsai/ci-imgs/assets/7608904/cbb6ebc0-1d07-49b2-bde5-099cbdaabed1">

Reply:
Yes, we'll need to exclude Ubuntu 18.04 for CUDA 12.2. Discussed here: https://github.com/rapidsai/shared-workflows/pull/166#discussion_r1439758006

Reply:
I attempted that exclusion in d51e8d926f470801da00e81973f0a8eea527fd59. Feel free to edit!

Reply:
To the original question from @ajschmidt8:

> I think these exclusions should be updated to include `12.2.2`.

I think we want:
- build on "newest CUDA minor version from each supported major version" = (11.8, 12.2)
- test on "newest CUDA minor version from previous major version" = 11.8; "oldest CUDA minor version from current major version" = 12.0, and "newest CUDA minor version (that RAPIDS supports) from current major version" = 12.2.

The testing part aligns with my comment here about previous policy discussions around the time we introduced CUDA 12.0 support: https://github.com/rapidsai/shared-workflows/pull/166#discussion_r1439743881

(We need to figure out how to document our _reasons_ and not just our results...)

Reply:
> I attempted that exclusion in https://github.com/rapidsai/ci-imgs/commit/d51e8d926f470801da00e81973f0a8eea527fd59

Works for me, thanks for that!

> I think we want...

@bdice Given this, do you think I could safely remove cuda11.4 builds from this repo?

Reply:
> do you think I could safely remove cuda11.4 builds from this repo?

Yes, probably, but I'd like to check with the ops team and separate that into a new PR.

Reply:
Ok great, I'll do that.

Reply:
eh well, maybe that's not as simple as I thought. I'll defer dropping CUDA 11.4 for now.

I see it's still tested against.

https://github.com/rapidsai/shared-workflows/blob/91799a905608f4c57d7fe65b92ce9a261249e635/.github/workflows/conda-cpp-tests.yaml#L66

https://github.com/rapidsai/shared-workflows/blob/91799a905608f4c57d7fe65b92ce9a261249e635/.github/workflows/conda-cpp-tests.yaml#L69

https://github.com/rapidsai/shared-workflows/blob/91799a905608f4c57d7fe65b92ce9a261249e635/.github/workflows/conda-python-tests.yaml#L74-L76

https://github.com/rapidsai/shared-workflows/blob/91799a905608f4c57d7fe65b92ce9a261249e635/.github/workflows/conda-python-tests.yaml#L74-L76

Reply:
Although this PR has already been merged, I'm following up for the record since I was tagged a handful of times. We ended up with mostly the right thing here. The current matrix is fine, but we could add some extra exclusions to avoid building unnecessary images. In particular, we could exclude CUDA 12.0/12.1 ci-wheel images since we will always be building on the latest (12.2), and we can probably exclude CUDA 12.1 citestwheel images since we only need to test earliest and latest.

Reply:
I'd wait to drop 12.0 ci-wheel images until we complete https://github.com/rapidsai/build-planning/issues/7.

I agree we don't need CUDA 12.1 images for anything, those should be safe to drop.

Reply:
>  I'd wait to drop 12.0 ci-wheel images until we complete https://github.com/rapidsai/build-planning/issues/7.

This is what I was thinking too. Most projects still haven't been cut over to building on 12.2 and therefore need the 12.0 images for building.

Reply:
Yup all good, I was making these comments from the perspective of this PR having already been merged and what future changes we would want to see.